google-cloud-spanner/src/main/java/com/google/cloud/spanner/SessionPool.java

```diff
@@ -2501,6 +2579,46 @@ private void removeLongRunningSessions(
         }
       }
     }
+
+    void maintainMultiplexedSession(Instant currentTime) {
+      try {
+        if (options.getUseMultiplexedSession()) {
+          synchronized (lock) {
+            if (multiplexedSession.get() != null && isMultiplexedSessionStale(currentTime)) {
+              logger.log(
+                  Level.INFO,
+                  String.format(
+                      "Removed Multiplexed Session => %s created at => %s",
+                      multiplexedSession.get().getName(),
+                      multiplexedSession.get().getDelegate().getCreateTime()));
+              if (multiplexedSessionRemovedListener != null) {
+                multiplexedSessionRemovedListener.apply(multiplexedSession.get());
```

Comment:
This should not be called here, as the existing session is not yet being removed.

Reply:
Removed it from here and added to `onSessionReady` method.